components/network/setup.ftl

```diff
@@ -99,27 +119,38 @@
               }
           /]
         [/#if]
-
         [#local routeTable = getLinkTarget(occurrence, networkLink + { "RouteTable" : routeTableId }, false)]
         [#local routeTableZones = routeTable.State.Resources["routeTables"]]
 
-        [#local networkACL = getLinkTarget(occurrence, networkLink + { "NetworkACL" : networkACLId }, false)]
-        [#local networkSecurityGroupZones = networkACL.State.Resources["networkSecurityGroups"]]
-
         [#list zones as zone]
 
           [#if zoneSubnets[zone.Id]?has_content]
 
             [#local zoneSubnetResources = zoneSubnets[zone.Id]]
-            [#local subnetName = zoneSubnetResources["subnet"].Name]
+            [#local subnetId = zoneSubnetResources["subnet"].Id]
+
+            [#local subnetName = formatAzureResourceName(
+              zoneSubnetResources["subnet"].Name,
+              AZURE_SUBNET_RESOURCE_TYPE,
+              vnetName)]
+
             [#local subnetAddress = zoneSubnetResources["subnet"].Address]
             [#local routeTableId = (routeTableZones[zone.Id]["routeTable"]).Id]
+            [#local routeTableName = (routeTableZones[zone.Id]["routeTable"]).Name]
 
             [@createSubnet
+              id=subnetId
               name=subnetName
+              vnetName=vnetName
               addressPrefix=subnetAddress
-              networkSecurityGroup={ "id" : networkSecurityGroupId }
-              routeTable= { "id" : routeTableId }     
+              networkSecurityGroup={ "id" : getReference(networkSecurityGroupId, networkSecurityGroupName) }
+              routeTable= { "id" : getReference(routeTableId, routeTableName) }     
+              dependsOn=
+                [
+                  getReference(vnetId, vnetName),
+                  getReference(networkSecurityGroupId, networkSecurityGroupName),
+                  getReference(routeTableId, routeTableName)
```

Comment:
Does Azure have implicit dependencies? I.e. when you do a get reference does it automatically add the dependency?

Reply:
This is using the Azure provider's getReference() function. It determines if the reference exists within the current template, and if so utilises ARM Template functions to format a reference to that resource (and sometimes attributes of that resource). 

In your example where its being used to populate an array of dependencies, it will result in a string like this: 
```"[resourceId('Microsoft.Network/virtualNetworks', 'djankywalks-integration-management-vpc')]"``` 
where the resourceId() ARM function is used to reference the yet-to-be-determined resourceId of the Vnet. 

So the long answer is yes its getting the dependencies, but here I'm being more explicit than implicit. 

https://github.com/codeontap/gen3-azure/blob/b00f6d9a20244f1399e7127592cfc8e95bf9057b/services/resource.ftl#L41-L45